softwareComponents/kinematics/calculations.hpp

```diff
@@ -9,331 +6,83 @@
-        }
-        inverse = rotation * translation;
-        return inverse;
-    }
-
-};
-
-struct joint {
-    double rx, tx, tz, rz;
-
-    joint() = default;
-    joint( double rx, double tx, double tz, double rz ) : rx( rx ), tx( tx ), tz( tz ), rz( rz ) { }
-
-    t_matrix to_previous(){
-        return t_matrix( rx, tx, tz, rz );
-    }
-    t_matrix from_previous(){
-        return t_matrix( rx, tx, tz, rz ).inv();
-    }
-};
-double constexpr error = 0.02;
-
-struct kinematic_chain {
-
-    Configuration config;
-
-    std::vector< int > ids; // fixed order for modules based on how they're connected;
-
-    std::vector< joint > chain; // chain of kinematic joints
-
-    std::vector< std::vector< double > > joint_positions;
-
-    std::vector< std::vector< double > > goals;
-
-    // Initialize the kinematic chain and positions
-    void init( std::vector< double > starting_pos = { 0.0, 0.0, 0.0, 0.0 }){
-        // TODO: More sophisticated way to choose the fix point, this could cause problems
-        ids = config.getIDs();
-        std::sort( ids.begin(), ids.end() );
-
-        chain.emplace_back( starting_pos[ 0 ],
-                            starting_pos[ 1 ],
-                            starting_pos[ 2 ],
-                            starting_pos[ 3 ] );
+inline Vector cross( const Vector& a, const Vector& b ){
+    return Vector({ a[ 1 ] * b[ 2 ] - a[ 2 ] * b[ 1 ],
+                    a[ 2 ] * b[ 0 ] - a[ 0 ] * b[ 2 ],
+                    a[ 0 ] * b[ 1 ] - a[ 1 ] * b[ 0 ] } );
+}
 
-        chain.emplace_back( to_rad( config.getModule( ids[ 0 ] ).getJoint( Joint::Alpha ) ),
-                            0,
-                            1,
-                            to_rad( config.getModule( ids[ 0 ] ).getJoint( Joint::Gamma ) ) );
+inline double scalar( const Vector& vector ){
+    return std::sqrt( std::pow( vector[ 0 ], 2 ) +
+                      std::pow( vector[ 1 ], 2 ) +
+                      std::pow( vector[ 2 ], 2 ) );
+}
 
-        for( size_t i = 1; i < ids.size(); ++i ){
-            chain.emplace_back( to_rad( config.getModule( ids[ i - 1 ] ).getJoint( Joint::Beta ) ),
-                                0,
-                                1,
-                                0 );
-            chain.emplace_back( to_rad( config.getModule( ids[ i ] ).getJoint( Joint::Alpha ) ),
-                                0,
-                                1,
-                                to_rad( config.getModule( ids[ i ] ).getJoint( Joint::Gamma ) ) );
-        }
-        // an imaginary joint at the end, helps us get the rotation/connection of two modules right
-        chain.emplace_back( 0, 0, 1, 0 );
+// dot
+inline double operator*( const Vector& a, const Vector& b ){
+    return a[ 0 ] * b[ 0 ] + a[ 1 ] * b[ 1 ] + a[ 2 ] * b[ 2 ];
+}
 
-        for( const auto& j : chain ){
-            joint_positions.push_back( { 0.0, 0.0, 0.0, 1.0 } );
+inline Matrix inverse( Matrix m ){
```

Comment:
I don't see how this computes an inverse matrix. If it does, it would be worth adding a comment here.